linkerd/service-profiles/src/split.rs

```diff
@@ -0,0 +1,195 @@
+use crate::{Profile, Receiver, Target};
+use futures::{prelude::*, ready};
+use indexmap::IndexSet;
+use linkerd2_addr::Addr;
+use linkerd2_error::Error;
+use linkerd2_stack::{layer, NewService};
+use rand::distributions::{Distribution, WeightedIndex};
+use rand::{rngs::SmallRng, SeedableRng};
+use std::{
+    marker::PhantomData,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower::ready_cache::ReadyCache;
+use tracing::debug;
+
+pub fn layer<N, S, Req>() -> impl layer::Layer<N, Service = NewSplit<N, S, Req>> + Clone {
+    let rng = SmallRng::from_entropy();
+    layer::mk(move |inner| NewSplit {
+        inner,
+        rng: rng.clone(),
```

Comment:
should we be doing something like [the second example here](https://github.com/linkerd/linkerd2-proxy/blob/bf0c222aef3a388d502f6cba5895e8efc48d92df/linkerd/service-profiles/src/http/service.rs#L32-L33)? i'll admit i'm not too familiar w/ `rand`'s APIs...

Reply:
@hawkw is that the right link? if so, i'm not sure what you're suggesting...

Reply:
whoops, it is not --- i meant to link to here: https://docs.rs/rand/0.7.3/rand/rngs/struct.SmallRng.html#examples